nixos/modules/services/misc/paperless.nix

```diff
@@ -439,5 +489,37 @@ in
         gid = config.ids.gids.paperless;
       };
     };
-  };
+  }) (lib.mkIf (cfg.enable && cfg.exporter.enable) {
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.exporter.directory}' - ${cfg.user} ${config.users.users.${cfg.user}.group} - -"
+    ];
+
+    systemd.timers.paperless-exporter = lib.mkIf (cfg.exporter.onCalendar != null) {
+      description = "paperless exporter timer";
+      wantedBy = [ "timers.target" ];
+      timerConfig = {
+        OnCalendar = cfg.exporter.onCalendar;
+        AccuracySec = "5m";
+      };
+    };
+
+    systemd.services.paperless-exporter = {
+      description = "paperless exporter service";
+      serviceConfig.User = cfg.user;
+      script = ''
+        set -e
+        cd ${cfg.dataDir}
+        echo "Running pre script ..."
+        ${cfg.exporter.postScript}
+
+        echo "Exporting documents ..."
```

Comment:
Nit: Scheduled service scripts logging stuff when nothing interesting is happening is a little pet peeve of mine. Systemd will already tell you that the service ran successfully.

Reply:
Yeah, I get that. However, I'm more a fan on the side of providing some output to be able to diagnose issues (like things getting silently stuck) and helping my future me debug things without being lost in a black hole.